Helper/Order.php

```diff
@@ -83,55 +83,12 @@ class Order extends AbstractHelper
     const BOLT_ORDER_STATUS_PENDING = 'bolt_pending';
     const MAGENTO_ORDER_STATUS_PENDING = 'pending';
 
-    // Posible transation state transitions
-    private $validStateTransitions = [
```

Comment:
so removing this means that we will now allow a pending transaction to go to auth state and vice versa. The last itme i tried to push this PR i didnt understand enough, but would like to to know why we are not adding to the states and rather nuking the state check?

cc @matt-thomason

Reply:
cc @daisy1754 because he had thoughts too

Reply:
https://stripe.com/docs/webhooks/best-practices#event-ordering

stripe works by asking clients to not assume ordering works.

Reply:
I agree with Matt, can we link to the hail changes in this thread? I would like to see the newly updated logic in its entirety.

Reply:
i think my question is, since we receive webhooks from hail (and though they are not out of order now, this could happen by chance?) and if that does happen and the plugin does whatever, isnt that bad? do we want to rely on webhook ordering 100%?

Reply:
lets just add some tests for the cases we care very much about and i think that should be good?

Reply:
There aren't any hail changes for this @ethanwayda22. I'm also not aware of what Asana tasks were created for this (it's kind of gotten ping-ponged to three different people).

Reply:
I added 3 main test cases:

- credit card first being authorized
- paypal on autocapture
- afterpay on complete, initially authorized

I think those are a good range of cases to start. I'm planning on adding a factory helper in a separate PR so we can start making testing easier on M2. I can fill in more test cases then for other functions to prove that the factories are effective, and go from there.